### PR TITLE
Relative timestamps

### PR DESCRIPTION
**Pull request by Pandabehr**

### CODE REVIEW

**Comment by Pandabehr** [comment]
Should probably add some javascript magic for the titles to be prettyfied on all browsers

**Comment by ChadTaljaardt** [comment]
Why not just use something like momentjs?

**Comment by range-of-motion** [comment]
For what? Displaying dates? I'd much rather do that on the server side..

**Comment by ChadTaljaardt** [comment]
@range-of-motion yeah, i always thought it was better to display dates in specific formats on the client side though i haven't really thought of the pros and cons of it, just always thought it should be a frontend thing. 

Can i know your reasoning as to why it's better done backend?

**Comment by range-of-motion** [comment]
@ChadTaljaardt Sure. In my opinion the front-end should be used for "moving parts". Parts like date pickers, charts and such. I think it'd be bad practice to load an entire library on the front-end just to format a date. Besides, you're adding weight to whatever the client has to fetch and process.

File: resources/lang/dk/calendar.php [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
     ],
     'weekdays' => [
-        0 => 'Mandag',
-        1 => 'Tirsdag',
-        2 => 'Onsdag',
-        3 => 'Torsdag',
-        4 => 'Fredag',
-        5 => 'Lørdag',
-        6 => 'Søndag'
+        1 => 'Mandag',
```
Comment (jardayn) [inline resources/lang/dk/calendar.php]: But... why?

File: app/Helpers/Format.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+{
+    public static function relativeTimeFormat($input_date)
+    {
+        $ts = strtotime($input_date);
+        $diff = strtotime('00:05:00') - $ts;
+
+        $translator = app('translator');
+
+
+        if ($diff >= 0) {
```
Comment (jardayn) [inline app/Helpers/Format.php]: Why not use something more simple?
http://php.net/manual/en/datetime.diff.php

File: app/Earning.php [hunk 3]
```diff
@@ -18,9 +19,7 @@ public function getFormattedAmountAttribute() {
     }
 
     public function getFormattedHappenedOnAttribute() {
-        $secondsDifference = strtotime(date('Y-m-d')) - strtotime($this->happened_on);
-
-        return ($secondsDifference / 60 / 60 / 24) . ' days ago';
+        return Format::relativeTimeFormat($this->happened_on);
```
Comment (jardayn) [inline app/Earning.php]: Copy pasting code. Should be put in a trait.

File: app/Earning.php [hunk 4]
```diff
@@ -4,11 +4,12 @@
 
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\SoftDeletes;
+use App\Helpers\Format;
 
 class Earning extends Model {
     use SoftDeletes;
 
-    protected $dates = ['deleted_at'];
+    protected $dates = ['deleted_at', 'happened_on'];
```
Comment (jardayn) [inline app/Earning.php]: Why is there no date_created or date_modified in here?

File: app/Helpers/Format.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Helpers;
+
+class Format
```
Comment (jardayn) [inline app/Helpers/Format.php]: Format is too abstract. Format What? DateFormatter? Formatter?

File: app/Helpers/Format.php [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+{
+    public static function relativeTimeFormat($input_date)
```
Comment (jardayn) [inline app/Helpers/Format.php]: why not require a DateTime object?

**Pull request closed without merging**